KH2: Update 00battle, 03system and objentry editors

Rikux3 opened the pull request:
Massive update on the source code files, as the documentation is much more advanced now.

The docs are now newer than the source code, so the tools should be updated at some point.

Rikux3 commented:
This PR is getting too big, so i'm putting this up for review now. 00battle and 00objentry documents should also be compliant with our documentation guidelines now. The rest can be done later.
@osdanova @1234567890num if you wouldn't mind to look over this (and of course everyone else who wants to)

osdanova commented on docs/kh2/file/type/jiminy.md:
```diff
@@ -120,28 +130,40 @@ Magic Code `JMAN`
+| 06     | byte   | Draw
+| 07     | byte   | Padding 
 
 Currently unknown. Read in z_un_002a18d0.
```
I take it this line was for "structure 2", which is now known.

osdanova commented on docs/kh2/file/type/00objentry.md:
```diff
@@ -6,76 +6,111 @@ Contains a definition of every object and it's parameters.
+| 0x5C   | uint16 | Spawn additional object 3 - [OBJ LIST](../../dictionary/obj.md)
+| 0x5E   | uint16 | Spawn additional object 4 - [OBJ LIST](../../dictionary/obj.md)
+
+*The first bit is somewhat related to the enemy state. Eg: Undead Pirates lose their immunity if they are changed from 2 to 0
```
This line is no longer needed

osdanova commented on OpenKh.Tools.Kh2BattleEditor/ViewModels/PlrpViewModel.cs:
```diff
@@ -0,0 +1,104 @@
+        //        _searchTerm = value;
+        //        PerformFiltering();
+        //    }
+        //}
```
Guess these commented out pieces of code are non functional yet. I don't think it should be merged unless you plan on implementing it in the near future.

osdanova reviewed (changes requested):
Tools work as intended. Some changes I suggest before merging:

Enmp: Update the fields in the view.
![image](https://user-images.githubusercontent.com/20492864/112719755-c0a5ed80-8efa-11eb-98c6-0a0ae4813788.png)

Bons: Remove "Unknown", which is padding.
![image](https://user-images.githubusercontent.com/20492864/112719764-d2879080-8efa-11eb-90f7-88b2d57030ba.png)

Sumn: Nothing is shown in the list, which is confusing. Can we add something there? Like the command's Id. Also, the field "ID" doesn't exist.
![image](https://user-images.githubusercontent.com/20492864/112719811-21cdc100-8efb-11eb-8f4c-3dab9dfcbf31.png)

Objentry: Since you addded flag checkboxes, the "Flag" field should be removed.
![image](https://user-images.githubusercontent.com/20492864/112719868-956fce00-8efb-11eb-8be8-294eadbd651b.png)

osdanova commented on docs/kh2/file/type/00battle.md:
```diff
@@ -650,17 +732,19 @@ Contains the level-up table for summons and drive forms.
+| 00     | uint8 : 0-3 | [Id](#forms)
+| 00     | uint8 : 4-7 | Level
+| 01     | uint8 : 0-3 | Ability Level
+| 01     | uint8 : 4-7 | Anti Rate
```
Anti rate and Ability Level are inversed. (Anti = 0-3; Abi = 4-7)

1234567890num replied:
What are Ability Levels?

osdanova replied:
Growth Ability levels for each form.
Eg: Valor form's level 1 and 2 have a Lvl 1 High Jump.

Rikux3 replied:
![grafik](https://user-images.githubusercontent.com/29046361/112736809-bfee7500-8f55-11eb-859f-fb63d74245a1.png)
Am i confusing something?

osdanova replied:
I'll show you Valor data so you can see:
![image](https://user-images.githubusercontent.com/20492864/112747683-3d969d00-8fb7-11eb-8fac-99638625d4cb.png)

And you wrote it right for the tool (Note that the function writes from right to left, so "4,4" are the first 4 bits):
![image](https://user-images.githubusercontent.com/20492864/112747690-47b89b80-8fb7-11eb-86c9-ab6d0701b0d7.png)
![image](https://user-images.githubusercontent.com/20492864/112747697-5010d680-8fb7-11eb-80f5-a43e126d1845.png)
![image](https://user-images.githubusercontent.com/20492864/112747705-5737e480-8fb7-11eb-8e36-b88856bcefcc.png)
![image](https://user-images.githubusercontent.com/20492864/112747710-5c952f00-8fb7-11eb-97c5-7ea088a2a184.png)

osdanova reviewed (approved):
Docs are alright and the updated tools work as intended. Looks good to me.

Rikux3 commented:
I've updated 03system's item table according to the decompilation. I would like for someone to verify this because it was confusing me a bit. I've also fixed the swapped events.
@1234567890num could you elaborate what you mean with the memt table?

1234567890num commented:
From ARD documentation:

#### Party

Set how the party member needs to be structured. According to `dbg/member.bin`, those are the only allowed values as a parameter:

| Value | Name          | Description
|-------|---------------|-------------
| 0     | NO_FRIEND     | Allow only Sora
| 1     | DEFAULT       | Allow Donald and Goofy
| 2     | W_FRIEND      | Allow Donald, Goofy and world guest.
| 3     | W_FRIEND_IN   | Same as `W_FRIEND` but forces the world guest to be in at the beginning.
| 4     | W_FRIEND_FIX  | Same as `W_FRIEND_IN`, but you can not swap the world guest.
| 5     | W_FRIEND_ONLY | Allow only with Sora and the world guest.
| 6     | DONALD_ONLY   | Only allow Sora and Donald.

The MEMT party table is what actually defines what these values do. For example, changing entry 0 of the table into 0x12030200 will give you Sora, Goofy, and Guest every time the game calls NO_FRIEND.

Rikux3 commented:
I would really like to merge this now.

Delta-473 commented:
Didn't had time to test it yet, but LGTM also needs rebase for the merge conflict.

Rikux3 commented:
I have resolved the issues on 00battle.

Delta-473 commented:
Can you look into why the tests broke, you might need to update those if you changed some names.

Rikux3 commented:
Closing this in favor of #489
